### DISCUSSION TITLE
ArrayItems 删除触发了重新渲染

### DISCUSSION THREAD

**Discussion by xinshangshangxin** [alibaba/formily#3451] [Q&A 2.x]
https://codesandbox.io/s/infallible-jerry-iqsm6e?file=/App.tsx

1. 将 第一个 删除
2. 这时候会看到 2,3,4 都重新渲染, 因为传入的value 全部变更

Q: 是不是 ArrayItems 用 index 作为 key, 导致删除的时候复用了组件? 应该如何优化才能 不复用组件, 删除1 下面的 2/3/4依然不重新渲染

PS: 拖拽也会有相同问题

**Comment by liweina-fans** [comment]
亲，问题解决了么，遇到同问题🥺
